[Style] Use strong Style value types for inset, margin and padding

weinig opened the pull request:
#### 4a9be0d0de8fae73bf03cb6f4ed25f65fa53a87f
<pre>
[Style] Use strong Style value types for inset, margin and padding
<a href="https://bugs.webkit.org/show_bug.cgi?id=293907">https://bugs.webkit.org/show_bug.cgi?id=293907</a>

Reviewed by Darin Adler.

Adds types to represent inset, margins and padding:

    Style::InsetEdge (wraps WebCore::Length)
    Style::InsetBox (minimally serializing group of Style::InsetEdges)

    Style::MarginEdge (wraps WebCore::Length)
    Style::MarginBox (minimally serializing group of Style::MarginEdges)

    Style::PaddingEdge (wraps WebCore::Length)
    Style::PaddingBox (minimally serializing group of Style::PaddingEdges)

The edge types are purely zero-cost wrappers of `WebCore::Length`, meaning there
should be no runtime overhead here. The goal of the types is to enforce what
keywords (e.g. &apos;auto&apos;) and value ranges (all or nonnegative) are allowed.

Additionally, instead of allowing direct access to the underlying `WebCore::Length`
or to `WebCore::Length::value()`, the numeric value can be accessed via functions
like `tryFixed()` which return `std::optional&lt;Style::Length&lt;R&gt;&gt;`. This means the
old idiomatic way for getting the value:

```
    if (edge.isFixed())
        useValue(edge.value());
```

is now replaced with:

```
    if (auto fixedValue = edge.tryFixed())
        useValue(fixedValue-&gt;value);
```

This ensures that the numeric value is not used without checking the type.

To make these new types possible, a few infrastructure changes were needed:

- `Style::Blending` was enhanced to add a new `requiresInterpolationForAccumulativeIteration`
  function to allow delegating `Style::Interpolation::WrapperBase::requiresInterpolationForAccumulativeIteration`.
- `Style::Blending` was enhanced so that `canBlend` and the new `requiresInterpolationForAccumulativeIteration`
  are optional and have default values if not provided.
- `Style::Blending` was enhanced so that the dispatch functions now check which versions of
  the functions are implemented on the specializations allowing types to only implement the
  most advanced version they actually need.
- `Theme::controlPadding` was moved to `RenderTheme` to allow use of `Style::PaddingBox`.
  It is only called from `RenderTheme`.

There is quite a bit of duplication between the `*Edge` types that can and should be reduced
in follow on changes.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/animation/ViewTimeline.cpp:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/values/CSSValueAggregates.h:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/layout/Verification.cpp:
* Source/WebCore/layout/formattingContexts/FormattingGeometry.cpp:
* Source/WebCore/layout/formattingContexts/FormattingGeometry.h:
* Source/WebCore/layout/formattingContexts/block/BlockMarginCollapse.cpp:
* Source/WebCore/layout/integration/LayoutIntegrationBoxGeometryUpdater.cpp:
* Source/WebCore/layout/layouttree/LayoutTreeBuilder.cpp:
* Source/WebCore/page/PrintContext.cpp:
* Source/WebCore/page/scrolling/ScrollSnapOffsetsInfo.cpp:
* Source/WebCore/platform/Length.h:
* Source/WebCore/platform/LengthFunctions.cpp:
* Source/WebCore/platform/LengthFunctions.h:
* Source/WebCore/platform/RectEdges.h:
* Source/WebCore/platform/Theme.cpp:
* Source/WebCore/platform/Theme.h:
* Source/WebCore/platform/mac/ThemeMac.h:
* Source/WebCore/platform/mac/ThemeMac.mm:
* Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp:
* Source/WebCore/rendering/PositionedLayoutConstraints.cpp:
* Source/WebCore/rendering/PositionedLayoutConstraints.h:
* Source/WebCore/rendering/RenderBlock.cpp:
* Source/WebCore/rendering/RenderBlockFlow.cpp:
* Source/WebCore/rendering/RenderBox.cpp:
* Source/WebCore/rendering/RenderBoxModelObject.cpp:
* Source/WebCore/rendering/RenderBoxModelObject.h:
* Source/WebCore/rendering/RenderBoxModelObjectInlines.h:
* Source/WebCore/rendering/RenderDeprecatedFlexibleBox.cpp:
* Source/WebCore/rendering/RenderElement.cpp:
* Source/WebCore/rendering/RenderFlexibleBox.cpp:
* Source/WebCore/rendering/RenderFlexibleBox.h:
* Source/WebCore/rendering/RenderGrid.cpp:
* Source/WebCore/rendering/RenderInline.cpp:
* Source/WebCore/rendering/RenderListMarker.cpp:
* Source/WebCore/rendering/RenderMenuList.cpp:
* Source/WebCore/rendering/RenderReplaced.cpp:
* Source/WebCore/rendering/RenderScrollbarPart.cpp:
* Source/WebCore/rendering/RenderTable.cpp:
* Source/WebCore/rendering/RenderTextInlines.h:
* Source/WebCore/rendering/RenderTheme.cpp:
* Source/WebCore/rendering/RenderTheme.h:
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.h:
* Source/WebCore/rendering/ios/RenderThemeIOS.h:
* Source/WebCore/rendering/ios/RenderThemeIOS.mm:
* Source/WebCore/rendering/mac/RenderThemeMac.h:
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/rendering/style/StyleSurroundData.cpp:
* Source/WebCore/rendering/style/StyleSurroundData.h:
* Source/WebCore/style/StyleAdjuster.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/style/StyleExtractorCustom.h:
* Source/WebCore/style/StyleExtractorSerializer.h:
* Source/WebCore/style/StyleInterpolationFunctions.h:
* Source/WebCore/style/StyleInterpolationWrappers.h:
* Source/WebCore/style/values/StyleValueTypes.h:
* Source/WebCore/style/values/borders/StyleBoxShadow.cpp:
* Source/WebCore/style/values/borders/StyleBoxShadow.h:
* Source/WebCore/style/values/borders/StyleCornerShapeValue.h:
* Source/WebCore/style/values/box/StyleMargin.cpp: Added.
* Source/WebCore/style/values/box/StyleMargin.h: Added.
* Source/WebCore/style/values/box/StylePadding.cpp: Added.
* Source/WebCore/style/values/box/StylePadding.h: Added.
* Source/WebCore/style/values/color/StyleColor.cpp:
* Source/WebCore/style/values/color/StyleDynamicRangeLimit.h:
* Source/WebCore/style/values/position/StyleInset.cpp: Added.
* Source/WebCore/style/values/position/StyleInset.h: Added.
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+Blending.h:
* Source/WebCore/style/values/scroll-snap/StyleScrollMargin.cpp:
* Source/WebCore/style/values/scroll-snap/StyleScrollMargin.h:
* Source/WebCore/style/values/scroll-snap/StyleScrollPadding.cpp:
* Source/WebCore/style/values/scroll-snap/StyleScrollPadding.h:
* Source/WebCore/style/values/text-decoration/StyleTextShadow.cpp:
* Source/WebCore/style/values/text-decoration/StyleTextShadow.h:

Canonical link: <a href="https://commits.webkit.org/295835@main">https://commits.webkit.org/295835@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ac886bfcef7265c56b7079c62d83dff485ecfefb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/106284 "28 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/26033 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/16429 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/111482 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/56880 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/108323 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/26696 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/34536 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/80734 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/109288 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21128 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/95914 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61061 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/20647 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14012 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/56320 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/90477 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14047 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/114342 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/33422 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/24646 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/89806 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/33786 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/92142 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/89507 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22825 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/34384 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12197 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/29006 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/33347 "Built successfully") | [  ~~🛠 mac-safer-cpp~~](https://ews-build.webkit.org/#/builders/120/builds/38759 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/33093 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/36446 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/34691 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/css/values/primitives/CSSPrimitiveNumericUnits.h:
```diff
@@ -55,6 +55,18 @@ struct ValueLiteral {
 
+    // Support unary operators.
+
+    constexpr ValueLiteral operator+()
```
Surprised that anyone writes code that relies on unary `+`.

weinig replied:
I figured since I needed - I might as well add +.

darinadler commented on Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
```diff
@@ -323,16 +325,21 @@ void RenderThemeAdwaita::adjustMenuListButtonStyle(RenderStyle& style, const Ele
+        return { 0_css_px };
 
     auto zoomedArrowSize = menuListButtonArrowSize * style.usedZoom();
     int leftPadding = menuListButtonPadding + (style.writingMode().isBidiRTL() ? zoomedArrowSize : 0);
     int rightPadding = menuListButtonPadding + (style.writingMode().isBidiLTR() ? zoomedArrowSize : 0);
```
Seems like these should just be `float`.

weinig replied:
I am hesitant to change something so subtle like that in this big a change.

darinadler commented on Source/WebCore/rendering/mac/RenderThemeMac.mm:
```diff
@@ -1064,14 +1066,16 @@ static void setFontFromControlSize(RenderStyle& style, NSControlSize controlSize
+            Style::Length<CSS::Nonnegative> { static_cast<float>(static_cast<int>(padding[topPadding] * style.usedZoom())) },
+            Style::Length<CSS::Nonnegative> { static_cast<float>(static_cast<int>(padding[rightPadding] * style.usedZoom())) },
+            Style::Length<CSS::Nonnegative> { static_cast<float>(static_cast<int>(padding[bottomPadding] * style.usedZoom())) },
+            Style::Length<CSS::Nonnegative> { static_cast<float>(static_cast<int>(padding[leftPadding] * style.usedZoom())) },
```
Surprised that we want this style of truncation. I do see that this most likely preserves the old behavior, but it doesn’t seem quite right. Would `std::floor` give us the same result in a less roundabout way?

Also, how about a helper function to make this less repetitive?

weinig replied:
Added helpers, but went with std::trunc to match match existing cast-to-c semantics.

darinadler commented on Source/WebCore/rendering/mac/RenderThemeMac.mm:
```diff
@@ -1080,13 +1084,16 @@ static void setFontFromControlSize(RenderStyle& style, NSControlSize controlSize
+            Style::Length<CSS::Nonnegative> { static_cast<float>(static_cast<int>(styledPopupPaddingTop * style.usedZoom())) },
+            Style::Length<CSS::Nonnegative> { static_cast<float>(static_cast<int>(rightPadding)) },
+            Style::Length<CSS::Nonnegative> { static_cast<float>(static_cast<int>(styledPopupPaddingBottom * style.usedZoom())) },
+            Style::Length<CSS::Nonnegative> { static_cast<float>(static_cast<int>(leftPadding)) },
```
Surprised that we want this style of truncation. I do see that this most likely preserves the old behavior, but it doesn’t seem quite right. Would `std::floor` give us the same result in a less roundabout way?

Also, how about a helper function or two to make this less repetitive?

darinadler commented on Source/WebCore/rendering/mac/RenderThemeMac.mm:
```diff
@@ -1190,11 +1197,7 @@ static void setFontFromControlSize(RenderStyle& style, NSControlSize controlSize
-    style.setPaddingRight(Length(padding, LengthType::Fixed));
-    style.setPaddingTop(Length(padding, LengthType::Fixed));
-    style.setPaddingBottom(Length(padding, LengthType::Fixed));
+    style.setPaddingBox({ Style::Length<CSS::Nonnegative> { static_cast<float>(static_cast<int>(1 * style.usedZoom())) } });
```
Surprised that we want this style of truncation. I do see that this most likely preserves the old behavior, but it doesn’t seem quite right. Would `std::floor` give us the same result in a less roundabout way?

Also, how about a helper function to make this less repetitive?

darinadler commented on Source/WebCore/rendering/mac/RenderThemeMac.mm:
```diff
@@ -1319,6 +1322,23 @@ static void setFontFromControlSize(RenderStyle& style, NSControlSize controlSize
+        // by definition constrained, since we select mini only for small cramped environments).
+        // This also guarantees the HTML <button> will match our rendering by default, since we're using
+        // a consistent padding.
+        auto length = Style::Length<CSS::Nonnegative> { static_cast<float>(static_cast<int>(8 * zoomFactor)) };
```
Surprised that we want this style of truncation. I do see that this most likely preserves the old behavior, but it doesn’t seem quite right. Would `std::floor` give us the same result in a less roundabout way?

Also, how about a helper function to make this less repetitive?

darinadler commented on Source/WebCore/style/StyleAdjuster.cpp:
```diff
@@ -110,16 +110,16 @@ static void addIntrinsicMargins(RenderStyle& style)
         if (style.marginBottom().hasQuirk())
-            style.setMarginBottom(WebCore::Length(intrinsicMargin, LengthType::Fixed));
+            style.setMarginBottom(Style::Length<> { static_cast<float>(intrinsicMargin) });
     }
```
How about making `intrinsicMargin` be some kind of `Length` or a `float` instead of an `int`?

weinig committed:
[Style] Use strong Style value types for inset, margin and padding
https://bugs.webkit.org/show_bug.cgi?id=293907

Reviewed by Darin Adler.

Adds types to represent inset, margins and padding:

    Style::InsetEdge (wraps WebCore::Length)
    Style::InsetBox (minimally serializing group of Style::InsetEdges)

    Style::MarginEdge (wraps WebCore::Length)
    Style::MarginBox (minimally serializing group of Style::MarginEdges)

    Style::PaddingEdge (wraps WebCore::Length)
    Style::PaddingBox (minimally serializing group of Style::PaddingEdges)

The edge types are purely zero-cost wrappers of `WebCore::Length`, meaning there
should be no runtime overhead here. The goal of the types is to enforce what
keywords (e.g. 'auto') and value ranges (all or nonnegative) are allowed.

Additionally, instead of allowing direct access to the underlying `WebCore::Length`
or to `WebCore::Length::value()`, the numeric value can be accessed via functions
like `tryFixed()` which return `std::optional<Style::Length<R>>`. This means the
old idiomatic way for getting the value:

```
    if (edge.isFixed())
        useValue(edge.value());
```

is now replaced with:

```
    if (auto fixedValue = edge.tryFixed())
        useValue(fixedValue->value);
```

This ensures that the numeric value is not used without checking the type.

To make these new types possible, a few infrastructure changes were needed:

- `Style::Blending` was enhanced to add a new `requiresInterpolationForAccumulativeIteration`
  function to allow delegating `Style::Interpolation::WrapperBase::requiresInterpolationForAccumulativeIteration`.
- `Style::Blending` was enhanced so that `canBlend` and the new `requiresInterpolationForAccumulativeIteration`
  are optional and have default values if not provided.
- `Style::Blending` was enhanced so that the dispatch functions now check which versions of
  the functions are implemented on the specializations allowing types to only implement the
  most advanced version they actually need.
- `Theme::controlPadding` was moved to `RenderTheme` to allow use of `Style::PaddingBox`.
  It is only called from `RenderTheme`.

There is quite a bit of duplication between the `*Edge` types that can and should be reduced
in follow on changes.

* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/animation/ViewTimeline.cpp:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/values/CSSValueAggregates.h:
* Source/WebCore/dom/Document.cpp:
* Source/WebCore/layout/Verification.cpp:
* Source/WebCore/layout/formattingContexts/FormattingGeometry.cpp:
* Source/WebCore/layout/formattingContexts/FormattingGeometry.h:
* Source/WebCore/layout/formattingContexts/block/BlockMarginCollapse.cpp:
* Source/WebCore/layout/integration/LayoutIntegrationBoxGeometryUpdater.cpp:
* Source/WebCore/layout/layouttree/LayoutTreeBuilder.cpp:
* Source/WebCore/page/PrintContext.cpp:
* Source/WebCore/page/scrolling/ScrollSnapOffsetsInfo.cpp:
* Source/WebCore/platform/Length.h:
* Source/WebCore/platform/LengthFunctions.cpp:
* Source/WebCore/platform/LengthFunctions.h:
* Source/WebCore/platform/RectEdges.h:
* Source/WebCore/platform/Theme.cpp:
* Source/WebCore/platform/Theme.h:
* Source/WebCore/platform/mac/ThemeMac.h:
* Source/WebCore/platform/mac/ThemeMac.mm:
* Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp:
* Source/WebCore/rendering/PositionedLayoutConstraints.cpp:
* Source/WebCore/rendering/PositionedLayoutConstraints.h:
* Source/WebCore/rendering/RenderBlock.cpp:
* Source/WebCore/rendering/RenderBlockFlow.cpp:
* Source/WebCore/rendering/RenderBox.cpp:
* Source/WebCore/rendering/RenderBoxModelObject.cpp:
* Source/WebCore/rendering/RenderBoxModelObject.h:
* Source/WebCore/rendering/RenderBoxModelObjectInlines.h:
* Source/WebCore/rendering/RenderDeprecatedFlexibleBox.cpp:
* Source/WebCore/rendering/RenderElement.cpp:
* Source/WebCore/rendering/RenderFlexibleBox.cpp:
* Source/WebCore/rendering/RenderFlexibleBox.h:
* Source/WebCore/rendering/RenderGrid.cpp:
* Source/WebCore/rendering/RenderInline.cpp:
* Source/WebCore/rendering/RenderListMarker.cpp:
* Source/WebCore/rendering/RenderMenuList.cpp:
* Source/WebCore/rendering/RenderReplaced.cpp:
* Source/WebCore/rendering/RenderScrollbarPart.cpp:
* Source/WebCore/rendering/RenderTable.cpp:
* Source/WebCore/rendering/RenderTextInlines.h:
* Source/WebCore/rendering/RenderTheme.cpp:
* Source/WebCore/rendering/RenderTheme.h:
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.cpp:
* Source/WebCore/rendering/adwaita/RenderThemeAdwaita.h:
* Source/WebCore/rendering/ios/RenderThemeIOS.h:
* Source/WebCore/rendering/ios/RenderThemeIOS.mm:
* Source/WebCore/rendering/mac/RenderThemeMac.h:
* Source/WebCore/rendering/mac/RenderThemeMac.mm:
* Source/WebCore/rendering/style/RenderStyle.cpp:
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleInlines.h:
* Source/WebCore/rendering/style/RenderStyleSetters.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:
* Source/WebCore/rendering/style/StyleSurroundData.cpp:
* Source/WebCore/rendering/style/StyleSurroundData.h:
* Source/WebCore/style/StyleAdjuster.cpp:
* Source/WebCore/style/StyleBuilderConverter.h:
* Source/WebCore/style/StyleBuilderCustom.h:
* Source/WebCore/style/StyleExtractorConverter.h:
* Source/WebCore/style/StyleExtractorCustom.h:
* Source/WebCore/style/StyleExtractorSerializer.h:
* Source/WebCore/style/StyleInterpolationFunctions.h:
* Source/WebCore/style/StyleInterpolationWrappers.h:
* Source/WebCore/style/values/StyleValueTypes.h:
* Source/WebCore/style/values/borders/StyleBoxShadow.cpp:
* Source/WebCore/style/values/borders/StyleBoxShadow.h:
* Source/WebCore/style/values/borders/StyleCornerShapeValue.h:
* Source/WebCore/style/values/box/StyleMargin.cpp: Added.
* Source/WebCore/style/values/box/StyleMargin.h: Added.
* Source/WebCore/style/values/box/StylePadding.cpp: Added.
* Source/WebCore/style/values/box/StylePadding.h: Added.
* Source/WebCore/style/values/color/StyleColor.cpp:
* Source/WebCore/style/values/color/StyleDynamicRangeLimit.h:
* Source/WebCore/style/values/position/StyleInset.cpp: Added.
* Source/WebCore/style/values/position/StyleInset.h: Added.
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+Blending.h:
* Source/WebCore/style/values/scroll-snap/StyleScrollMargin.cpp:
* Source/WebCore/style/values/scroll-snap/StyleScrollMargin.h:
* Source/WebCore/style/values/scroll-snap/StyleScrollPadding.cpp:
* Source/WebCore/style/values/scroll-snap/StyleScrollPadding.h:
* Source/WebCore/style/values/text-decoration/StyleTextShadow.cpp:
* Source/WebCore/style/values/text-decoration/StyleTextShadow.h:

Canonical link: https://commits.webkit.org/295835@main
